tests/models/graphormer/test_modeling_graphormer.py

```diff
@@ -0,0 +1,1545 @@
+            check_hidden_states_output(inputs_dict, config, model_class)
+
+    def test_retain_grad_hidden_states_attentions(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        config.output_hidden_states = True
+        config.output_attentions = False
+
+        # no need to test all models as different heads yield the same functionality
+        model_class = self.all_model_classes[0]
+        model = model_class(config)
+        model.to(torch_device)
+
+        outputs = model(**inputs_dict)
+        output = outputs[0]
+
+        hidden_states = outputs.hidden_states[0]
+        hidden_states.retain_grad()
+
+        output.flatten()[0].backward(retain_graph=True)
+
+        self.assertIsNotNone(hidden_states.grad)
+
+    # Inputs are 'input_nodes' and 'input_edges' not 'input_ids'
+    def test_model_main_input_name(self):
+        for model_class in self.all_model_classes:
+            model_signature = inspect.signature(getattr(model_class, "forward"))
+            # The main input is the name of the argument after `self`
+            observed_main_input_name_nodes = list(model_signature.parameters.keys())[1]
+            observed_main_input_name_edges = list(model_signature.parameters.keys())[2]
+            self.assertEqual(model_class.main_input_name_nodes, observed_main_input_name_nodes)
+            self.assertEqual(model_class.main_input_name_edges, observed_main_input_name_edges)
+
+    def test_forward_signature(self):
+        config, _ = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+            signature = inspect.signature(model.forward)
+            # signature.parameters is an OrderedDict => so arg_names order is deterministic
+            arg_names = [*signature.parameters.keys()]
+
+            expected_arg_names = ["input_nodes", "input_edges"]
+            self.assertListEqual(arg_names[:2], expected_arg_names)
+
+    def test_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model(*config_and_inputs)
+
+    def test_for_graph_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_graph_classification(*config_and_inputs)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in GRAPHORMER_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = GraphormerForGraphClassification.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+
+@require_torch
+class GraphormerModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_graph_classification(self):
+        model = GraphormerForGraphClassification.from_pretrained("graphormer-base-pcqm4mv2")
+
+        # Actual real graph data from the MUTAG dataset
```

Comment:
Let's ignroe formatting for this tensor so it takes less vertical space.

Reply:
Is there an "ignore format" tag I should add for the tests?

Reply:
`# fmt: off` on the line above and `# fmt: on` on the line below.